src/auth/auth.controller.ts

```diff
@@ -60,6 +92,16 @@ export class AuthController {
       });
     }
 
+    if (req.csrf) {
+      const token = this.getCsrfTokenFromCookies(cookie);
+      if (token !== req.csrf) {
+        throw new UnauthorizedException({
+          error: 'Invalid credentials',
+          location: __filename,
+        });
+      }
+    }
```

Comment:
`login` is not responsible for CSRF token verification. You should move it to a separate middleware

Reply:
In middleware we have no access to body so I moved it to guard